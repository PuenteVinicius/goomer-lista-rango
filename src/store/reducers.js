import { combineReducers } from 'redux'
import gridRestaurantsReducer from '../containers/grid-restaurants/gridRestaurantsReducer';
import searchrestaurantsReducer from '../containers/search-restaurants/searchrestaurantsReducer';
import menuReducer from '../containers/menu/menuReducer';

const reducers = combineReducers({
  gridRestaurants: gridRestaurantsReducer,
  searchRestaurants: searchrestaurantsReducer,
  menu: menuReducer 
})

export default reducers;