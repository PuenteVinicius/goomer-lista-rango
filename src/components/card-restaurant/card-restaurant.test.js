import React from "react";
import toJson from "enzyme-to-json";
import promise from 'redux-promise'
import multi from 'redux-multi'
import thunk from 'redux-thunk'
import { mount } from "enzyme";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../../store/reducers";
import CardRestaurant from "./card-restaurant";

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__ 
      && window.__REDUX_DEVTOOLS_EXTENSION__()
const store = applyMiddleware(multi, thunk, promise)(createStore)(reducers, devTools)

it("should render the homepage", () => {
  const props = {
    initialValue: "Seu nome aqui"
  };
  const cardRestaurant = mount(
    <Provider store={store}>
      <CardRestaurant {...props} />
    </Provider>
  );
  expect(toJson(cardRestaurant)).toMatchSnapshot();
});
