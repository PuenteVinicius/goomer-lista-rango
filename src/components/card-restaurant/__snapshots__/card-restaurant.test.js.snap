// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the homepage 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CardRestaurant)
    initialValue="Seu nome aqui"
  >
    <CardRestaurant
      changeValue={[Function]}
      initialValue="Seu nome aqui"
      value="Seu nome aqui"
    >
      <div />
    </CardRestaurant>
  </Connect(CardRestaurant)>
</Provider>
`;
